src/app/Caches/Create/CreateCacheWizard.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { useEffect, useState } from 'react';
+import { Wizard } from '@patternfly/react-core';
+import GettingStarted from './GettingStarted';
+import EditCache from './EditConfig';
+import { useHistory } from 'react-router';
+import { useApiAlert } from '@app/utils/useApiAlert';
+import { ConsoleServices } from '@services/ConsoleServices';
+import { CacheConfigUtils } from '@services/cacheConfigUtils';
+import { useTranslation } from 'react-i18next';
+
+const CreateCacheWizard = (props) => {
+    const { addAlert } = useApiAlert();
+
+    const [formCacheName, setFormCacheName] = useState('')
+    const [formCreateType, setFormCreateType] = useState<'configure' | 'edit'>('edit');
+    const [formConfig, setFormConfig] = useState('')
+    const [stepIdReached, setStepIdReached] = useState(1);
+    const [isFormValid, setIsFormValid] = useState(false);
+    const [configs, setConfigs] = useState<OptionSelect[]>([]);
+    const [validConfig, setValidConfig] = useState<
+        'success' | 'error' | 'default'
+    >('default');
+    const [errorConfig, setErrorConfig] = useState('');
+
+    const history = useHistory();
+
+    interface OptionSelect {
+        value: string;
+        disabled?: boolean;
+        isPlaceholder?: boolean;
+    }
+
+    const title = 'Create Cache';
+
+    const onFormChange = (isValid, value) => {
+        setIsFormValid(isValid);
+        setFormCacheName(value.cacheName);
+        setFormCreateType(value.createType);
+    };
+
+    const onFormChangeConfig = (config) => {
+        setFormConfig(config)
+    }
+
+    const createCache = () => {
+        const name = formCacheName.trim();
+
+        // Validate Name
+        const isValidName: 'success' | 'error' = name.length > 0 ? 'success' : 'error';
+
+        // Validate the config
+        let isValidConfig: 'success' | 'error';
+        let configValidation;
+
+        if (configs.length == 0) {
+            // there are no  templates or the expanded area is opened, we validate the text area content
```

Comment:
create cache here is dealing two things. One of them is creating the cache using a template. This is how it works today. You can leave this behaviour and modify the create cache second step with it (even if it's not in the design right now). To understand how this works, try to run the server with a template defined (check the xml and the it tests) and you will see how it's working today. We can totally keep that behaviour. 

However, what needs to be done is having the parent wizard component keep the state and pass the method that changes the state to the step child component. otherwise, this will never work